appeals/api/src/server/endpoints/integrations/integrations.broadcasters/appeal.js

```diff
@@ -1,65 +1,117 @@
-	// 		appellantCase: {
-	// 			include: {
-	// 				appellantCaseInvalidReasonsSelected: {
-	// 					include: {
-	// 						appellantCaseInvalidReason: true,
-	// 						appellantCaseInvalidReasonText: true
-	// 					}
-	// 				},
-	// 				appellantCaseIncompleteReasonsSelected: {
-	// 					include: {
-	// 						appellantCaseIncompleteReason: true,
-	// 						appellantCaseIncompleteReasonText: true
-	// 					}
-	// 				},
-	// 				knowsAllOwners: true,
-	// 				knowsOtherOwners: true
-	// 			}
-	// 		},
-	// 		lpaQuestionnaire: {
-	// 			include: {
-	// 				lpaQuestionnaireIncompleteReasonsSelected: {
-	// 					include: {
-	// 						lpaQuestionnaireIncompleteReason: true,
-	// 						lpaQuestionnaireIncompleteReasonText: true
-	// 					}
-	// 				}
-	// 			}
-	// 		}
-	// 	}
-	// });
+	const appeal = await databaseConnector.appeal.findUnique({
+		where: { id: appealId },
+		include: {
+			address: true,
+			appealTimetable: true,
+			procedureType: true,
+			appealType: true,
+			appealStatus: true,
+			caseOfficer: true,
+			inspector: true,
+			lpa: true,
+			allocation: true,
+			siteVisit: true,
+			inspectorDecision: true,
+			specialisms: {
+				include: {
+					specialism: true
+				}
+			},
+			parentAppeals: true,
+			childAppeals: true,
+			appellantCase: {
+				include: {
+					appellantCaseValidationOutcome: true,
+					appellantCaseInvalidReasonsSelected: {
+						include: {
+							appellantCaseInvalidReason: true,
+							appellantCaseInvalidReasonText: true
+						}
+					},
+					appellantCaseIncompleteReasonsSelected: {
+						include: {
+							appellantCaseIncompleteReason: true,
+							appellantCaseIncompleteReasonText: true
+						}
+					},
+					knowsAllOwners: true,
+					knowsOtherOwners: true
+				}
+			},
+			lpaQuestionnaire: {
+				include: {
+					lpaQuestionnaireValidationOutcome: true,
+					lpaQuestionnaireIncompleteReasonsSelected: {
+						include: {
+							lpaQuestionnaireIncompleteReason: true,
+							lpaQuestionnaireIncompleteReasonText: true
+						}
+					}
+				}
+			}
+		}
+	});
 
-	//console.log(appeal)
+	if (!appeal) {
+		pino.error(`Trying to broadcast info for appeal ${appealId} , but it was not found.`);
+		return false;
+	}
+
+	const appealRelationships = [...appeal.parentAppeals, ...appeal.childAppeals];
+
+	const linkedAppeals = appealRelationships.filter(
+		(relationship) => relationship.type === CASE_RELATIONSHIP_LINKED
+	);
+	const relatedAppeals = appealRelationships.filter(
+		(relationship) => relationship.type === CASE_RELATIONSHIP_RELATED
+	);
+
+	// @ts-ignore
+	const msg = messageMappers.mapAppeal({
+		...appeal,
+		linkedAppeals,
+		relatedAppeals
+	});
+	if (msg) {
+		const validationResult = await validateFromSchema(schemas.events.appeal, msg);
+		if (validationResult !== true && validationResult.errors) {
+			const errorDetails = validationResult.errors?.map(
+				(e) => `${e.instancePath || '/'}: ${e.message}`
+			);
+
+			pino.error(`Error validating appeal: ${errorDetails[0]}`);
```

Comment:
Will we setup alerts or something for this error? Else it'd be a silent error and appeal may never be broadcast?

Reply:
yeah is a good point...

Reply:
@beejjacobs I have an approval here - I would merge and look at this later, if you are happy with the PR generally...

Reply:
@fabfab-kc  yes it's worth 1) making a ticket so we don't forget, 2) add some fields to make it easier and provide more info, use a static message field and the log query will be easier e.g

`pino.error({appealId: appeal.id, //...}, 'error validating appeal for broadcast');`

do we definitely not want to throw this error? That would at least alert the user?